src/types/streams/converted_source.rs

```diff
@@ -0,0 +1,26 @@
+use url::Url;
+
+use crate::types::{resource::StreamSource, torrent::InfoHash};
+
+/// Trait which defines the StreamSource state data structures in Core.
+pub trait StreamSourceTrait: sealed::Sealed {}
+/// only we should be able to define which data structures are StreamSource states!
+mod sealed {
+    pub trait Sealed {}
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ConvertedStreamSource {
```

Comment:
I'm not sure about having this struct, seems like a redundant one duplicating properties. Since The conversion end result is an url maybe just having a `ResourceLoadable<Url>` in player model would be enough, or have some specific deeplinks method which would be async and would add the url to the `deeplinks.externalLink.streaming` or smth?

Reply:
We talked about this in Slack, tldr is that we want to have explicit state conversions, this is not used but I'll make sure to slowly transition core to this state-transition logic to make it explicit.